src/public/modules/forms/helpers/ndjsonStream.ts

```diff
@@ -0,0 +1,72 @@
+// Modified fork of https://github.com/canjs/can-ndjson-stream to enqueue
+// the string immediately without a JSON.parse() step, as the stream payload
+// is to be decrypted by the decryption worker.
+
+// Note that this code assumes a polyfill of TextDecoder is available to run in IE11.
+
+export const ndjsonStream = (
+  response: ReadableStream<Uint8Array>,
+): ReadableStream => {
+  // For cancellation
+  let maybeReader: ReadableStreamDefaultReader<Uint8Array> | undefined
+  let shouldCancel = false
+  return new ReadableStream({
+    start: function (controller) {
+      const reader = response.getReader()
+      maybeReader = reader
+      const decoder = new TextDecoder()
+      let data_buf = ''
+
+      return reader
+        .read()
+        .then(function processResult(result): Promise<void> | undefined | void {
+          if (result.done && shouldCancel) {
+            return
+          }
+
+          if (result.done) {
+            data_buf = data_buf.trim()
+            if (data_buf.length !== 0) {
+              try {
+                controller.enqueue(data_buf)
+              } catch (e) {
+                controller.error(e)
+                return
+              }
+            }
+            return controller.close()
+          }
+
+          // Read the input in as a stream and split by newline and trim
+          data_buf += decoder.decode(result.value, { stream: true })
+          const lines = data_buf.split('\n')
+          const readableLines = lines.slice(0, -1).map((line) => line.trim())
+          const bufferStore = lines[lines.length - 1]
```

Comment:
Should we keep this part the same? We are conscious of performance in this file, and slicing then mapping incurs an additional `O(n) + O(n)` cost. Should we literally just copy and paste the original file and add types? All these refactors are actually reducing the optimization already done in the original library.

Reply:
I understand that performance is a key factor in this file and given that this is the case, maybe it's reasonable to change it back to the old version.

i'll add comments so that this part is easier to parse and understand!